packages/components/src/KaizenProvider/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react"
+import { defaultTheme, Theme } from "@kaizen/design-tokens"
+import { ThemeManager } from "./ThemeManager"
+
+export const ThemeContext = React.createContext<Theme>(defaultTheme)
+
+/**
+ * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
+ * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
+ */
+export const ThemeProvider = ({
+  themeManager,
+  ...props
+}: {
+  themeManager?: ThemeManager
+  children: React.ReactNode
+}): JSX.Element => {
+  const [themeManagerInstance] = useState(
```

Comment:
Nitpick: I'd like if we could add the generic to the `useState<SomethingHere>` if possible :)